static/js/services/enketo.js

```diff
@@ -109,7 +116,18 @@ angular.module('inboxServices').service('Enketo',
           })
           .then(transformXml);
       }
-      return xmlCache[id][language];
+      return $q.resolve()
+        .then(function() {
+          return xmlCache[id][language];
+        })
+        .then(function(doc) {
+          // clone doc to avoid leaking of data between instances of a form
+          return {
+            html: doc.html.clone(),
+            model: doc.model,
+            hasContactSummary: doc.hasContactSummary
```

Comment:
`doc` is probably the wrong name here. It's just a cache entry that stores everything we need to render the form because transforming the xml is expensive so we only do it once. I felt it was cleaner to store detect the `contact-summary` instance using a selector so it has to be done before transform, so it has to be stored in the cache too. Does that make sense?